From 41d0941b160817d917eeffc66f89334dd2e1243e Mon Sep 17 00:00:00 2001
From: xiaoqiang <xiaoqiang@xq-desktop.(none)>
Date: Fri, 10 Dec 2010 11:47:12 +0800
Subject: [PATCH] multi-ver package should be erased from notupdated list

Packages, which support mult-version install, should not be in
notupdated list, or zypper would report some conflicting messages when
updating system.

See reference BUG: http://bugs.meego.com/show_bug.cgi?id=1595
---
 src/Summary.cc |    9 +++++++++
 1 files changed, 9 insertions(+), 0 deletions(-)

diff --git a/src/Summary.cc b/src/Summary.cc
index 684f334..a435095 100644
--- a/src/Summary.cc
+++ b/src/Summary.cc
@@ -313,6 +313,15 @@ void Summary::readPool(const zypp::ResPool & pool)
         inserter(notupdated[*kit], notupdated[*kit].begin()),
         Summary::ResPairNameCompare());
 
+  for_(it, notupdated.begin(), notupdated.end())
+  {
+    for_(resit, it->second.begin(), it->second.end())
+    {
+      if (resit->second->multiversionInstall() == true)
+        it->second.erase(resit);
+    }
+  }
+
   // remove kinds with empty sets after the set_difference
   for (KindToResPairSet::iterator it = notupdated.begin(); it != notupdated.end();)
   {
-- 
1.7.2.2

